src/vm/diagnosticprotocolhelper.cpp

```
@@ -88,4 +93,137 @@ void DiagnosticProtocolHelper::GenerateCoreDump(IpcStream* pStream)
+        // the header, etc, the realistic max is around 4000 characters.
+        hr = E_INVALIDARG;
+        goto ErrExit;
+    }
+
+    pwszProfilerPath = new (nothrow) WCHAR[cbProfilerPath];
+    if (pwszProfilerPath == nullptr)
+    {
+        hr = E_INVALIDARG;
+        goto ErrExit;
+    }
+
+    pathSize = cbProfilerPath * sizeof(WCHAR);
+    memcpy(pwszProfilerPath, pBufferCursor, pathSize);
+    bufferLen -= pathSize;
+    pBufferCursor += pathSize;
+        
+    if (!TryParse(pBufferCursor, bufferLen, cbClientData))
+    {
+        hr = E_INVALIDARG;
+        goto ErrExit;
+    }
+
+    pClientData = new (nothrow) uint8_t[cbClientData];
+    if (pClientData == nullptr)
+    {
+        hr = E_OUTOFMEMORY;
+        goto ErrExit;
+    }
+
+    bufferBytesRead = 0;
+    pClientDataCursor = pClientData;
+    // TODO: get rid of this ad-hoc byte[] parsing code
+    while (bufferBytesRead < cbClientData)
+    {
+        if (bufferLen == 0)
+        {
+            // Client data was bigger than the buffer, need to read more
+            fSuccess = pStream->Read(buffer, sizeof(buffer), nNumberOfBytesRead);
+            if (!fSuccess)
+            {
+                hr = E_UNEXPECTED;
+                goto ErrExit;
+            }
+
+            pBufferCursor = buffer;
+            bufferLen = nNumberOfBytesRead;
+        }
+
+        bytesToRead = min((cbClientData - bufferBytesRead), bufferLen);
+        memcpy(pClientDataCursor, pBufferCursor, bytesToRead);
+        pClientDataCursor += bytesToRead;
+        
+        _ASSERTE(bytesToRead <= bufferLen && "bytesToRead > bufferLen means we overran the buffer");
+        bufferLen -= bytesToRead;
+        bufferBytesRead += bytesToRead;
+    }
+
+    _ASSERTE(bufferBytesRead == cbClientData && "bufferBytesRead > cbClientData means we read too far");
+
+    if (cbClientData == 0)
+    {
+        pClientData = nullptr;
+    }
+
+    hr = ProfilingAPIUtility::LoadProfilerForAttach(&profilerGuid,
```

The diagnostics server starts up fairly early in the runtime initialization, prior to ProfilingAPIUtility::InitializeProfiling() having been called. We should return an error if we haven't initialized the profiling sub-system yet or arrange for this thread to block until that initialization has completed.